app/src/ui/lib/parse-conflict-modify-delete-files.ts

```diff
@@ -0,0 +1,46 @@
+/** Parse CONFLICT (modify/delete) errors to get the files that caused the error
+ *
+ * Example Error:
+ * CONFLICT (modify/delete): app/src/ui/lib/draggable.tsx deleted in HEAD and modified in 0f777848d (Allow switching tab in branch menu during drag). Version 0f777848d (Allow switching tab in branch menu during drag) of app/src/ui/lib/draggable.tsx left in tree.
+ * Auto-merging app/src/ui/branches/branches-container.tsx
+ * CONFLICT (modify/delete): app/src/lib/drag-and-drop-manager.ts deleted in HEAD and modified in 0f777848d (Allow switching tab in branch menu during drag). Version 0f777848d (Allow switching tab in branch menu during drag) of app/src/lib/drag-and-drop-manager.ts left in tree.
+ */
+
+export function parseConflictModifyDeleteFiles(
+  stdOut: string
+): ReadonlyArray<string> {
+  const files = new Array<string>()
+  const lines = stdOut.split('\n')
+
+  for (const line of lines) {
+    if (!line.startsWith('CONFLICT')) {
+      continue
+    }
+
+    const splitOnSpace = line.split(' ')
+    if (splitOnSpace.length >= 3) {
+      files.push(splitOnSpace[2].trim())
+    }
+  }
+
+  return files
+}
+
+/**
+ * Parse CONFLICT (modify/delete) errors to get commit summary of the commit
+ * that caused the error
+ */
+export function parseConflictModifyDeleteCommitSummary(stdOut: string): string {
+  const versionRegex = /Version\s.*\((.*)\)/
```

Comment:
I think this regex causes issues if the commit summary has `()` in it:

![image](https://user-images.githubusercontent.com/1083228/112473879-34b38a80-8d6f-11eb-96f7-2fbafe57d10c.png)

I think this should work:
```suggestion
  const versionRegex = /Version\s[^(]*\((.*)\)/
```

https://regex101.com/ is very useful to test this stuff 😄 

Another issue I found is if the commit summary has a `Version whatever (...` in it, since that commit summary is also printed _before_ the `Version` you're using as reference:
![image](https://user-images.githubusercontent.com/1083228/112474347-cb804700-8d6f-11eb-98fe-b91fe0e2d4a0.png)

I think we'll need a more complex logic for this parser 😰

Reply:
Awesome catches and yep I love that site.